src/System.Windows.Forms.Design.Editors/src/System/Drawing/Design/FontEditor.cs

```
@@ -11,53 +11,42 @@
 namespace System.Drawing.Design
 {
     /// <summary>
-    ///       Provides a font editor that
-    ///       is used to visually select and configure a Font
-    ///       object.
+    /// Provides a font editor that is used to visually select and configure a Font object.
     /// </summary>
     [CLSCompliant(false)]
     public class FontEditor : UITypeEditor
     {
-        private FontDialog fontDialog;
-        private object value;
+        private FontDialog _fontDialog;
 
         /// <summary>
-        ///      Edits the given object value using the editor style provided by
-        ///      GetEditorStyle.  A service provider is provided so that any
-        ///      required editing services can be obtained.
+        /// Edits the given object value using the editor style provided by GetEditorStyle.
+        /// A service provider is provided so that any required editing services can be obtained.
         /// </summary>
         public override object EditValue(ITypeDescriptorContext context, IServiceProvider provider, object value)
         {
-            this.value = value;
-
-            Debug.Assert(provider != null, "No service provider; we cannot edit the value");
             if (provider != null)
             {
-                IWindowsFormsEditorService edSvc = (IWindowsFormsEditorService)provider.GetService(typeof(IWindowsFormsEditorService));
```

Isn't that a common idiom that a service for an interface implements that interface? I've always considered it a bug in the caller if he provided a bogus service provider, not being worth programming defensively against.

You'll definitively have a lot of places to "fix" this, I'd assume most GetService calls are a strict cast and not an as-cast with fallback code

---

I agree with @weltkante, this is too large a change and the paradigm is too spread out. Please undo this line change

---

I don't agree here so I'll try to explain what my logic is.
Across winforms we have about a 50-50 split between using `(X)provider.GetService(typeof(X))` and using `provider.GetService(typeof(x)) as X`. I think the use of `as/is` has numerous benefits
- Clears up confusion as to what happens when the service is `null` -  will this throw `InvalidCastException` or `NullReferenceException`. Actually, it's neither - it returns `null` - but the `as/is` pattern avoids this confusion.
- Most importantly, we can use the C# construct `if (variable is Type variableAsType)` which lets us made code more readable, express intent better etc.

It's essentially my view is not that we want to throw `InvalidCastException` - if this were the case we'd use the force cast rather than the `as` cast across winforms, which we don't. The intent to get an object of that type. In each case we check for null so we understand that the provider can return an invalid service. In some cases we've accounted for where the provider returns a non-interface service, in others we haven't. It's my opinion that we should standardise on the latter.

---

@hughbe my suggestion is to 
1. remove this change from this PR
1. create an issue discussing the discrepancy in the code base, your suggested change, and its benefits
1. await a decision from them team after broader community discussion

and then if the decision is to change these more broadly (I suspect we will want to align them all one way or the other), then that change should come in its own PR.

---

Just noting that previously we had a `Debug` assert here suggesting that we would never expect an invalid service to be returned from `provider.GetService`. In essence I've uncovered some edge cases that are highly unlikely to occur in practice, where the designer makes things valid. However, I still think this sort of test coverage is important :)))

```cs
Debug.Assert(provider != null, "No service provider; we cannot edit the value");
if (provider != null)
{
    IWindowsFormsEditorService edSvc = (IWindowsFormsEditorService)provider.GetService(typeof(IWindowsFormsEditorService));

    Debug.Assert(edSvc != null, "No editor service; we cannot edit the value");
    if (edSvc != null)
    {
        ...
    }
}
```

---

Now reverted the change, as per #817 decision